src/couch_jobs/test/couch_jobs_tests.erl

```diff
@@ -426,6 +429,40 @@ resubmit_enqueues_job(#{t1 := T, j1 := J}) ->
     end).
 
 
+resubmit_pending_updates_job_data(#{t1 := T, j1 := J}) ->
+    ?_test(begin
+        Data1 = #{<<"test">> => 1},
+        Data2 = #{<<"test">> => 2},
+        ok = couch_jobs:add(?TX, T, J, Data1),
+        ?assertEqual(ok, couch_jobs:add(?TX, T, J, Data2, 6)),
+        ?assertMatch({ok, _, Data2}, couch_jobs:accept(T))
+    end).
+
+
+resubmit_finished_updates_job_data(#{t1 := T, j1 := J}) ->
+    ?_test(begin
+        Data1 = #{<<"test">> => 1},
+        Data2 = #{<<"test">> => 2},
+        ok = couch_jobs:add(?TX, T, J, Data1),
+        {ok, Job1, #{}} = couch_jobs:accept(T),
+        ?assertEqual(ok, couch_jobs:finish(?TX, Job1)),
+        ?assertEqual(ok, couch_jobs:add(?TX, T, J, Data2, 6)),
+        ?assertMatch({ok, _, Data2}, couch_jobs:accept(T))
+    end).
+
+
+resubmit_running_does_not_update_job_data(#{t1 := T, j1 := J}) ->
+    ?_test(begin
+        Data1 = #{<<"test">> => 1},
+        Data2 = #{<<"test">> => 2},
+        ok = couch_jobs:add(?TX, T, J, Data1),
+        {ok, Job1, #{}} = couch_jobs:accept(T),
+        ?assertEqual(ok, couch_jobs:add(?TX, T, J, Data2, 6)),
+        ?assertEqual(ok, couch_jobs:finish(?TX, Job1)),
+        ?assertMatch({ok, _, Data1}, couch_jobs:accept(T))
+    end).
+
+
```

Comment:
Let's rename these to `add_...` since they go through `add/5` API only and also have `resubmit_` versions that go through `resubmit` API as well. So first is an `add/4,5` then we call `resubmit/4`. It may be redundant but it it is in case we change either API behavior in the future we'd want to see if the behavior changed for them individually.

For the resubmit cases also let's check how resubmit with `undefined` data works in those states.